[lldb] Skip TestVariableAnnotationsDisassembler.py on non-x86 architectures

UltimateForce21 opened the pull request:
The test lldb-api::TestVariableAnnotationsDisassembler.py was failing on the lldb-remote-linux-ubuntu and lldb-remote-linux-win builders due to assembler incompatibilities in d_original_example.s. These failures are not related to the disassembler changes themselves but to the test setup.

This patch updates the test to be skipped when running on unsupported architectures to avoid failures. The test will still run and validate correctly where the assembler input is supported.

tstellar commented:
I've tested this on aarch64 and it fixes the failure.

DavidSpickett reviewed (approved):
LGTM, thanks!

DavidSpickett commented:
Please adjust your email settings as described in https://llvm.org/docs/DeveloperPolicy.html#email-addresses, then this can be merged.

(I could push the commit on your behalf right now, but I don't want to do anything with your personal information without your permission)

UltimateForce21 committed:
[lldb] Skip TestVariableAnnotationsDisassembler.py on non-x86 architectures

The test TestVariableAnnotationsDisassembler.py relies on an .s input file generated for x86_64 with assembly directives that are not portable (e.g., DWARF sections and comments using #).
This causes build failures on non-x86 targets such as AArch64.

This change skips the test unless the architecture is x86_64, ensuring it only runs where the assembly input is valid.

DavidSpickett commented:
All good now, thanks.

> tried to amend the author and force push the same commit

Actually the way this repository is setup, we ignore what's in the commit aside from the changes of course. Final commit message comes from the PR description and the author details are governed by the author's GitHub settings.

But thank you for being diligent anyway!

slydiman commented:
This patch fixed
[lldb-remote-linux-ubuntu](https://lab.llvm.org/buildbot/#/builders/195/builds/13924)
and
[lldb-remote-linux-win](https://lab.llvm.org/buildbot/#/builders/197/builds/8541)

But the test `lldb-api::TestVariableAnnotationsDisassembler.py` is still broken on [lldb-x86_64-win](https://lab.llvm.org/buildbot/#/builders/211)

Please take a look.
